mx_fetchjdk.py

```diff
@@ -0,0 +1,213 @@
+        if args["strip-contents-home"]:
+            tmp_full_jdk_path = full_jdk_path + ".tmp"
+            shutil.move(full_jdk_path, tmp_full_jdk_path)
+            shutil.move(join(tmp_full_jdk_path, 'Contents', 'Home'), full_jdk_path)
+            shutil.rmtree(tmp_full_jdk_path)
+        else:
+            full_jdk_path = join(full_jdk_path, 'Contents', 'Home')
+
+    if "alias" in args:
+        alias_full_path = join(jdk_path, args["alias"])
+        if exists(alias_full_path) or islink(alias_full_path):
+            if isdir(alias_full_path) and not islink(alias_full_path):
+                shutil.rmtree(alias_full_path)
+            else:
+                os.remove(alias_full_path)
+
+        if not (mx.is_windows() or mx.is_cygwin()):
+            os.symlink(full_jdk_path, alias_full_path)
+        else:
+            copytree(full_jdk_path, alias_full_path, symlinks=True) # fallback for windows
+        full_jdk_path = alias_full_path
+
+    if not args["quiet"]:
+        print("Run the following to set JAVA_HOME in your shell:")
+
+    print('export JAVA_HOME={}'.format(abspath(full_jdk_path)))
+
+
+def _parse_fetchjdk_settings(args):
+    settings = {}
+    settings["quiet"] = False
+    settings["keep-archive"] = False
+    settings["jdk-path"] = default_jdk_path()
+
+    common_location = join(_mx_home, 'common.json')
+
+    parser = ArgumentParser(prog='mx fetch-jdk')
+    parser.add_argument('--java-distribution', action='store', help='JDK distribution that should be downloaded (e.g., "labsjdk-ce-11" or "openjdk8")')
+    parser.add_argument('--configuration', action='store', help='location of configuration json file (default: \'{}\')'.format(common_location))
+    parser.add_argument('--to', action='store', help='location where JDK would be downloaded (default: \'{}\')'.format(settings["jdk-path"]))
+    parser.add_argument('--alias', action='store', help='name of symlink to JDK')
+    parser.add_argument('--keep-archive', action='store_true', help='keep downloaded JDK archive')
+    if mx.is_darwin():
+        parser.add_argument('--strip-contents-home', action='store_true', help='strip Contents/Home')
+    parser.add_argument('-q', '--quiet', action='store_true', help='suppress logging output')
+    parser.add_argument('remainder', nargs=REMAINDER, metavar='...')
+    args = parser.parse_args(args)
+
+    if args.quiet is not None:
+        settings["quiet"] = args.quiet
+
+    if args.to is not None:
+        settings["jdk-path"] = args.to
+    elif not settings["quiet"]:
+        mx.warn("JDK location hasn't been specified. Using {}".format(settings["jdk-path"]))
+
+    if not check_access_jdk_path(settings["jdk-path"]):
+        warning_msg = "Path '{}' is not writable (try running with elevated privileges). ".format(settings["jdk-path"])
+        settings["jdk-path"] = abspath(join("bin", "jdks"))
+        warning_msg += os.linesep + "Trying to fall back to {}".format(settings["jdk-path"])
```

Comment:
Don't fall back. If the user does not want to install in the system standard location, then they must use the `--to` option. The preceding warning about the system standard location not being writable should be converted to an abort:
```
mx.abort("JDK installation directory {} is not writeable. Either re-run with elevated privileges (e.g. sudo) or specify a writeable directory with the --to option.")
```

Reply:
Fixed